soroban-env-host/src/budget.rs

```diff
@@ -111,33 +33,75 @@ impl CostType {
 /// The parameters for a `CostModel` are calibrated empirically. See this crate's
 /// benchmarks for more details.
 
-#[derive(Default, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct CostModel {
-    pub(crate) const_param: u64,
-    pub(crate) lin_param: u64,
+// #[derive(Default, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+// pub struct CostModel {
+//     pub(crate) const_param: u64,
+//     pub(crate) lin_param: u64,
+// }
+
+// impl CostModel {
+//     pub fn from_xdr(xdr: &ContractCostParamEntry) -> Self {
+//         CostModel {
+//             const_param: xdr.const_term as u64,
+//             lin_param: xdr.linear_term as u64,
+//         }
+//     }
+
+//     /// Evaluates the linear model `f(input) = const_param + lin_param * Option<input>`.
+//     /// If the input is `None`, then the linear part is ingored, resulting in a constant
+//     /// model.
+//     pub fn evaluate(&self, input: Option<u64>) -> u64 {
+//         match input {
+//             Some(input) => {
+//                 let mut res = self.const_param;
+//                 if self.lin_param != 0 {
+//                     res = res.saturating_add(self.lin_param.saturating_mul(input));
+//                 }
+//                 res
+//             }
+//             None => self.const_param,
+//         }
+//     }
+
+//     #[cfg(test)]
+//     pub fn reset(&mut self) {
+//         self.const_param = 0;
+//         self.lin_param = 0;
+//     }
+// }
+
+pub trait HostCostModel {
+    fn evaluate(&self, input: Option<u64>) -> Result<u64, HostError>;
+
+    #[cfg(test)]
+    fn reset(&mut self);
```

Comment:
I don't see anything using this `reset()` method (and it seems a little odd since it's just writing to public fields of a set of parameters that ought to never be changed -- not a tracker or budget limit). Can it be removed?

Reply:
The model `reset()` is being called by budget `reset_models()`, which is used in metering tests: create a Host, resets all models to zero, then set individual models of interest to the values we want, run the test and record the budget.

Reply:
ok